Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('publicIps') is not None:
+        kwargs.update({'PublicIps': parse_resource_ids(args.get('publicIps'))})
+    if args.get('allocationIds') is not None:
+        kwargs.update({'AllocationIds': parse_resource_ids(args.get('allocationIds'))})
+
+    response = client.describe_addresses(**kwargs)
+
+    for i, address in enumerate(response['Addresses']):
+        data.append({
+            'PublicIp': address['PublicIp'],
+            'AllocationId': address['AllocationId'],
+            'Domain': address['Domain'],
+            'Region': obj['_user_provided_options']['region_name'],
+        })
+        if 'InstanceId' in address:
+            data[i].update({'InstanceId': address['InstanceId']})
+        if 'AssociationId' in address:
+            data[i].update({'AssociationId': address['AssociationId']})
+        if 'NetworkInterfaceId' in address:
+            data[i].update({'NetworkInterfaceId': address['NetworkInterfaceId']})
+        if 'PrivateIpAddress' in address:
+            data[i].update({'PrivateIpAddress': address['PrivateIpAddress']})
+        if 'Tags' in address:
+            for tag in address['Tags']:
+                data[i].update({
+                    tag['Key']: tag['Value']
+                })
+
+    raw = response['Addresses']
+    raw[0].update({'Region': obj['_user_provided_options']['region_name']})
+    ec = {'AWS.EC2.ElasticIPs(val.AllocationId === obj.AllocationId)': raw}
+    human_readable = tableToMarkdown('AWS EC2 ElasticIPs', data)
+    return_outputs(human_readable, ec)
+
+
+def describe_snapshots_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('ownerIds') is not None:
+        kwargs.update({'OwnerIds': parse_resource_ids(args.get('ownerIds'))})
+    if args.get('snapshotIds') is not None:
+        kwargs.update({'SnapshotIds': parse_resource_ids(args.get('snapshotIds'))})
+    if args.get('restorableByUserIds') is not None:
+        kwargs.update({'RestorableByUserIds': parse_resource_ids(args.get('restorableByUserIds'))})
+
+    response = client.describe_snapshots(**kwargs)
+
+    for i, snapshot in enumerate(response['Snapshots']):
+        data.append({
+            'Description': snapshot['Description'],
+            'Encrypted': snapshot['Encrypted'],
+            'OwnerId': snapshot['OwnerId'],
+            'Progress': snapshot['Progress'],
+            'SnapshotId': snapshot['SnapshotId'],
+            'StartTime': datetime.strftime(snapshot['StartTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.